granulate_utils/metadata/cloud.py

```diff
@@ -151,6 +161,23 @@ def get_azure_metadata() -> Optional[AzureInstanceMetadata]:
     )
 
 
+def get_oracle_cloud_metadata() -> Optional[OracleCloudInstanceMetadata]:
+    # Documentation: https://docs.oracle.com/en-us/iaas/Content/Compute/Tasks/gettingmetadata.htm
+    response = send_request("http://169.254.169.254/opc/v2/instance/", headers={"Authorization": "Bearer Oracle"})
+    if response is None:
+        return None
+    instance = response.json()
+
+    return OracleCloudInstanceMetadata(
+        provider="oracle",
+        instance_type=instance["shape"],
+        region=instance["canonicalRegionName"],
+        memory=instance["shapeConfig"]["memoryInGBs"],
```

Comment:
Other cloud provider classes don't have this attribute (yet), let's see can get memory information on them as well.
Also, rename to include the memory units (`memory_mb` or `memory_gb`).